### PR TITLE
bpo-30420: Document the cwd argument to check_output and check_call

### PR DESCRIPTION
**Pull request by alex**

### CODE REVIEW

**Comment by vadmium** [comment]
There are other parameters missing compared to _Popen_, which is already mentioned in the documentation. Perhaps it would be better to add an ellipsis (...) or _**params_ at the end of the signature to make this more obvious.

**Comment by alex** [comment]
It's possible there are larger-scale changes that would be preferable. This was inspired by me using this parameter in some code, and @reaperhulk being skeptical that the parameter existed because it wasn't documented. This PR solves that problem, not broader challenges.

**Comment by ncoghlan** [comment]
+1 for adding `cwd` to the "Frequently Used Arguments" section: https://docs.python.org/3/library/subprocess.html#frequently-used-arguments

In addition to `check_call` and `check_output`, it should also be listed in the signatures for `call` and `run`.

Regarding the general concern, I started writing a longer reply here, and then realised I would be better off just filing an issue and posting the relevant details there: http://bugs.python.org/issue30420

Basically, I think we have pretty solid (albeit anecdotal) evidence that our current approach to making these docs less overwhelming is genuinely misleading people as to the available API options, and we need to tweak things accordingly.

**Comment by alex** [comment]
@ncoghlan good call (pun intended) on `call` and `run`

**Comment by ncoghlan** [comment]
This at least makes it clearer that `cwd` is accepted as an argument by the convenience APIs, so I went ahead and merged it. However, I'll move the issue back to patch needed, as while this is an improvement, I don't think it fully resolves the underlying problem.

**Comment by alex** [comment]
Sounds good.
